server.js

```diff
@@ -60,38 +64,53 @@ function deploy(config, ready) {
             }
         }
 
-        const stat = fs.statSync(requestedFile);
+        let stat = fs.statSync(requestedFile);
 
         if (stat.isDirectory()) {
-            if (!requestedFile.endsWith(path.sep)) {
-                requestedFile += path.sep;
-            }
-            response.writeHead(200, 'OK', { 'Content-Type': 'text/html' });
-            if (request.method === 'HEAD') {
-                response.end();
-                return;
-            }
-            response.write('<pre>\n');
-
-            let parentDir = path.resolve(path.normalize(path.join(requestedFile, '..')));
-            if (!parentDir.endsWith(path.sep)) {
-                parentDir += path.sep;
-            }
-            if (parentDir.startsWith(cwd)) {
-                let parentLink = '/' + toPosixPath(parentDir.slice(cwd.length));
-                if (parentLink === '/.') {
-                    parentLink = '/';
-                }
-                response.write(`<a href="${parentLink}">..</a>\n`);
-            }
-
-            for (const file of fs.readdirSync(requestedFile)) {
-                const fullPath = requestedFile + file;
-                response.write(`<a href="/${toPosixPath(fullPath.slice(cwd.length))}">${file}</a>\n`);
-            }
-            response.write('</pre>');
-            response.end();
-            return;
+			if (!requestedFile.endsWith(path.sep)) {
+				requestedFile += path.sep;
+			}
+			let indexFound = false;
+			config.IndexFiles.every(elem => {
+				let indexFile = requestedFile + 'index.html';
+				if(fs.existsSync(indexFile)){
+					requestedFile = indexFile;
+					stat = fs.statSync(requestedFile);
+					indexFound = true;
+					return false;
+				}
+				return true;
+			});
+
+			if(!indexFound) {
+				response.writeHead(200, 'OK', { 'Content-Type': 'text/html' });
+
+				if (request.method === 'HEAD') {
+					response.end();
+					return;
+				}
+				response.write('<pre>\n');
+
+				let parentDir = path.resolve(path.normalize(path.join(requestedFile, '..')));
+				if (!parentDir.endsWith(path.sep)) {
+					parentDir += path.sep;
+				}
+				if (parentDir.startsWith(cwd)) {
+					let parentLink = '/' + toPosixPath(parentDir.slice(cwd.length));
+					if (parentLink === '/.') {
+						parentLink = '/';
+					}
+					response.write(`<a href="${parentLink}">..</a>\n`);
+				}
+
+				for (const file of fs.readdirSync(requestedFile)) {
+					const fullPath = requestedFile + file;
+					response.write(`<a href="/${toPosixPath(fullPath.slice(cwd.length))}">${file}</a>\n`);
+				}
+				response.write('</pre>');
+				response.end();
+				return;
+			}
```

Comment:
1. Removed idention
2. utilized the `every` function result (https://jsfiddle.net/dnzmtfhu/)
3. changed
    ```
    let indexFile = requestedFile + "index.html";
    ```
    to
    ```
    const indexFile = requestedFile + elem;
    ```
 
```suggestion
	if (!requestedFile.endsWith(path.sep)) {
		requestedFile += path.sep;
	}
	const noIndexFound = config.indexFiles.every(elem => {
		const indexFile = requestedFile + elem;
		if(fs.existsSync(indexFile)){
			requestedFile = indexFile;
			stat = fs.statSync(requestedFile);
			return false;
		}
		return true;
	});

	if(noIndexFound) {
		response.writeHead(200, 'OK', { 'Content-Type': 'text/html' });

		if (request.method === 'HEAD') {
			response.end();
			return;
		}
		response.write('<pre>\n');

		let parentDir = path.resolve(path.normalize(path.join(requestedFile, '..')));
		if (!parentDir.endsWith(path.sep)) {
			parentDir += path.sep;
		}
		if (parentDir.startsWith(cwd)) {
			let parentLink = '/' + toPosixPath(parentDir.slice(cwd.length));
			if (parentLink === '/.') {
				parentLink = '/';
			}
			response.write(`<a href="${parentLink}">..</a>\n`);
		}

		for (const file of fs.readdirSync(requestedFile)) {
			const fullPath = requestedFile + file;
			response.write(`<a href="/${toPosixPath(fullPath.slice(cwd.length))}">${file}</a>\n`);
		}
		response.write('</pre>');
		response.end();
		return;
	}
```